### DISCUSSION TITLE
Record generic constraint

### DISCUSSION THREAD

**Discussion by pohorely** [dotnet/csharplang#4027] [General] [answered]
Is possible to constraint generic type to records in C# 9.0 somehow?

What I mean:
```csharp
public record MyRecord<T> where T : record
{
    public T MyProperty { get; set; }
}
```

**Comment by canton7** [comment]
What would a constraint of `record` give you, which a constraint of `class` does not?

**Comment by svick** [comment, marked as answer]
This is a duplicate of https://github.com/dotnet/csharplang/discussions/3859.
